packages/slate-editor/src/extensions/galleries/components/Gallery/SortableGalleryTile.tsx

```diff
@@ -0,0 +1,31 @@
+import type { UniqueIdentifier } from '@dnd-kit/core';
+import { useSortable } from '@dnd-kit/sortable';
+import React from 'react';
+
+import { GalleryTile, type Props } from './GalleryTile';
+
+export function SortableGalleryTile({
+    getIndex,
+    id,
+    ...props
+}: Props & { getIndex: (id: UniqueIdentifier) => number; id: UniqueIdentifier }) {
+    const { active, attributes, isDragging, listeners, over, setNodeRef } = useSortable({ id });
+
+    let insertPosition: 'before' | 'after' | undefined = undefined;
+    if (active && active.id !== id && over && over.id === id) {
+        const activeIndex = getIndex(active.id);
+        const index = getIndex(id);
+        insertPosition = activeIndex > index ? 'before' : 'after';
+    }
```

Comment:
Yeah, I was hoping there would be a simpler way to detect whether the "collision" happened on the edge and then depending on whether it's a left or right edge, I would show the "indicator". I think it will require manually detecting the collision via the bounding box. I would have to take a look at that.

There's another problem with edges though and that is, if you have one image on the edge of the row and then another image that's on the next row on the opposite edge. Due to the way our data is structured and how the algorithm works, it's for example impossible in this scenario to drag the image from 2nd row to the end of the 1st row and expect anything to happen (since the two images are already "next to each other" in the array).

From the user point of view, I would expect the image from the 2nd row to get "squeezed" to the end of the 1st row but that's unfortunately not how our layout algorithm works as it has no concept of rows, as the data structure is just a list. So, I think even if were able to show the "indicator" at the edge of the row, there's no guarantee that the image would be placed there as it's ultimately up to the algorithm to decide.

So, I am not sure how this can be improved without completely reworking the layout algorithm, which I haven't done for obvious reasons 🙂 

![Screenshot 2024-02-26 at 12 01 44](https://github.com/prezly/slate/assets/4209081/63fc1787-ae2a-4d77-b196-40b43b255ae3)

Reply:
Discussed this with @alpower and reworked the "drop" overlay to take the entire tile instead of appearing before/after. The drop behaviour is still the same, but it feels a bit more "natural" this way:

https://github.com/prezly/slate/assets/4209081/3d104066-86e5-4eca-ad0f-1ae0d1f6be0b